code/graph-algorithms/diameter_trees/cpp/Diameter.cpp

```diff
@@ -0,0 +1,65 @@
+//Learn O(n)
```

Comment:
Don't forget to add the required comment as stated in the [contribution guidelines](https://github.com/OpenGenus/cosmos/wiki/contribute).